src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -119,49 +122,43 @@
 
     private SqlFedAuthToken fedAuthToken = null;
 
-    static class Sha1HashKey {
-        private byte[] bytes;
+    static class CityHash128Key {
+        private long[] segments;
+        private int hashCode;
 
-        Sha1HashKey(String s) {
-            bytes = getSha1Digest().digest(s.getBytes());
+        CityHash128Key(String s) {
+            segments = CityHash.cityHash128(s.getBytes(), 0, s.length());
         }
 
         public boolean equals(Object obj) {
-            if (!(obj instanceof Sha1HashKey))
+            if (!(obj instanceof CityHash128Key))
                 return false;
 
-            return java.util.Arrays.equals(bytes, ((Sha1HashKey)obj).bytes);
+            return java.util.Arrays.equals(segments, ((CityHash128Key)obj).segments);
         }
 
         public int hashCode() {
-            return java.util.Arrays.hashCode(bytes);
-        }
-
-        private java.security.MessageDigest getSha1Digest() {
-            try {
-                return java.security.MessageDigest.getInstance("SHA-1");
-            }
-            catch (final java.security.NoSuchAlgorithmException e) {
-                // This is not theoretically possible, but we're forced to catch it anyway
-                throw new RuntimeException(e);
+            if (0 == hashCode) {
+                hashCode = java.util.Arrays.hashCode(segments); 
             }
+            return hashCode;
         }
     }
 
     /** Size of the parsed SQL-text metadata cache */
     static final private int PARSED_SQL_CACHE_SIZE = 100;
 
     /** Cache of parsed SQL meta data */
-    static private ConcurrentLinkedHashMap<Sha1HashKey, ParsedSQLCacheItem> parsedSQLCache;
+    static private ConcurrentLinkedHashMap<CityHash128Key, ParsedSQLCacheItem> parsedSQLCache;
 
     static {
-        parsedSQLCache = new Builder<Sha1HashKey, ParsedSQLCacheItem>()
+        parsedSQLCache = new Builder<CityHash128Key, ParsedSQLCacheItem>()
 	        .maximumWeightedCapacity(PARSED_SQL_CACHE_SIZE)
             .build();
     }
 
      /** Get prepared statement cache entry if exists, if not parse and create a new one */
-     static ParsedSQLCacheItem getOrCreateCachedParsedSQLMetadata(Sha1HashKey key, String sql) throws SQLServerException {
+     static ParsedSQLCacheItem getOrCreateCachedParsedSQL(CityHash128Key key, String sql) throws SQLServerException {
```

Comment:
@TobiasSQL I disagree here.  "Metadata" in the JDBC sense is typically either database metadata (``Connection.getMetaData()``), column metadata (``PreparedStatement.getParameterMetaData()``), or result metadata (``ResultSet.getMetaData()``).  In this case, the cache is simply the pre-processed SQL text of the statement, without respect to type information.